File: reactor/r2dbc-mysql/src/main/java/ua/mai/zyme/r2dbcmysql/repository/TransferRepository.java
package ua.mai.zyme.r2dbcmysql.repository;

import org.springframework.data.r2dbc.repository.R2dbcRepository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;
import ua.mai.zyme.r2dbcmysql.entity.Transfer;

public interface TransferRepository extends R2dbcRepository<Transfer, Long> {

  Mono<Transfer> findByTransferId(Long transferId);

  Flux<Transfer> findByFromMemberId(Integer fromMemberId);

  Flux<Transfer> findByToMemberId(Integer toMemberId);

  Flux<Transfer> findByFromMemberIdAndToMemberId(Integer fromMemberId, Integer toMemberId);

}

File: kotlin/kotlin-r2dbc-mysql/kotlin-r2dbc-mysql-be/src/main/java/ua/mai/zyme/r2dbcmysql/repository/TransferRepository.kt
package ua.mai.zyme.r2dbcmysql.repository

import org.springframework.data.r2dbc.repository.R2dbcRepository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import ua.mai.zyme.r2dbcmysql.entity.Transfer

interface TransferRepository : R2dbcRepository<Transfer, Long> {

    fun findByTransferId(transferId: Long): Mono<Transfer>

    fun findByFromMemberId(fromMemberId: Int): Flux<Transfer>

    fun findByToMemberId(toMemberId: Int): Flux<Transfer>

    fun findByFromMemberIdAndToMemberId(fromMemberId: Int, toMemberId: Int): Flux<Transfer>

}